server/src/languageServerBase.ts

```diff
@@ -507,6 +569,18 @@ export abstract class LanguageServerBase implements LanguageServerInterface {
             service.setFileClosed(filePath);
         });
 
+        this._connection.onDidChangeWatchedFiles(params => {
+            params.changes.forEach(change => {
+                const filePath = convertUriToPath(change.uri);
+                const eventType: FileWatcherEventType = change.type === 1 ? 'add' : 'change';
```

Comment:
Perhaps this is something to be left for later (as I'm noticing it's missing from previous enums like `FileWatcherEventType`), but do we not support full removals? I can see this coming up if a module gets deleted, which would probably have a `delete` event. Perhaps you're modelling that as a change and it works well enough.

Reply:
Chokidar and the node version don't differentiate between "change" and "delete" events. They're all reported as "change". I'm just normalizing everything here.

I'm not sure what you mean by "full removals"? Are you talking about when the root directory of a file watcher is deleted?

Reply:
Sorry, by "full" I meant "distinct from changed". I just didn't want to end up in a scenario where someone deletes a module but it remains resolvable.